### PR TITLE
[14.0][IMP] product_pricelist_assortment: sales administrator access

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
Let Sales administrator modify assortment items. In order to get it had to include sales_team module into dependency.
Before you had to have system admin to modify it, which is too much.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

**Comment by francesco-ooops** [comment]
@rousseldenis @acsonefho @Cedric-Pigeon good for you?

File: product_pricelist_assortment/security/product_pricelist_assortment_item.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <record model="ir.model.access" id="product_pricelist_assortment_item_manager">
         <field name="name">product.pricelist.assortment pricelist manager</field>
         <field name="model_id" ref="model_product_pricelist_assortment_item" />
-        <field name="group_id" ref="base.group_system" />
+        <field name="group_id" ref="sales_team.group_sale_manager" />
```
Comment (acsonefho) [inline product_pricelist_assortment/security/product_pricelist_assortment_item.xml:17]: I think adding `sales_team` just to fill a group is not a good idea.

Maybe create a dedicated group "Assortment manager" and then (into another module) link the Sales manager with Assortment manager.

**Review by acsonefho — CHANGES_REQUESTED** [review]

File: product_pricelist_assortment/security/product_pricelist_assortment_item.xml (hunk 1 above)
Comment (ilyasProgrammer) [reply product_pricelist_assortment/security/product_pricelist_assortment_item.xml:17]: Maybe. I'm not sure what would be more correct.

Comment (acsonefho) [reply product_pricelist_assortment/security/product_pricelist_assortment_item.xml:17]: I'm not in favor of adding `sales_team` if it's not necessary about functionnal behavior

Comment (francesco-ooops) [reply product_pricelist_assortment/security/product_pricelist_assortment_item.xml:17]: @acsonefho sorry this is not really clear to me - to use this module user needs to be able to edit a pricelist (Sales: administrator), so what would be the correct group to use the module?

Comment (francesco-ooops) [reply product_pricelist_assortment/security/product_pricelist_assortment_item.xml:17]: @acsonefho which group do you think is necessary for functional behavior?

Comment (acsonefho) [reply product_pricelist_assortment/security/product_pricelist_assortment_item.xml:17]: Oh sorry I miss read the module name. I thought that was `product_assortment`. It's ok for me :pray:

**Review by acsonefho — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@pedrobaeza good for a quick merge?

**Comment by pedrobaeza** [comment]
This change is a bit controversial, but if it's OK for one of the original contributors:

/ocabot merge major

**Comment by francesco-ooops** [comment]
> This change is a bit controversial, but if it's OK for one of the original contributors:

Can you expand on why? This module adds a feature to pricelist, and users who are allowed to manage pricelists must be in group "sales: administrator" (if I'm not mistaken). 

Also consider there is no restriction (to my knowledge) to assortment management, at least in v14

I don't see the reason for restricting pricelist assortment management to "administration: setting" group, but if there's a better solution that could have been implemented I'm all ears :)

**Comment by pedrobaeza** [comment]
- It's changing the dependencies on a stable module.
- It's changing who can access to the model.

**Comment by francesco-ooops** [comment]
I understand. On the other hand, I think this was an oversight in development stage rather than a security implementation, but I see what you mean, thanks for the clarification!

**Pull request merged**